test/test_typed_array.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "sparrow/typed_array.hpp"
+
+#include "doctest/doctest.h"
+
+#include <iostream> // For Doctest
+#include <numeric>
+
+using namespace sparrow;
+
+using data_type_t = int32_t;
+using my_typed_array = typed_array<data_type_t>;
+
+namespace
+{
+    array_data make_test_array_data(size_t n = 10, size_t offset = 0)
+    {
+        array_data ad;
+        ad.type = data_descriptor(data_type::INT32);
```

Comment:
This can be obtained from the trait via `type_id` now:

https://github.com/xtensor-stack/sparrow/blob/2a07414d2561f078a3f8878623a266608854afaa/include/sparrow/data_traits.hpp#L73